packages/shared/src/hooks/log/useLogContextData.ts

```diff
@@ -51,33 +51,42 @@ export default function useLogContextData(
   durationEventsQueue: MutableRefObject<Map<string, LogEvent>>,
   sendBeacon: () => void,
 ): LogContextData {
-  return useMemo<LogContextData>(
-    () => ({
-      logEvent(event: LogEvent) {
-        pushToQueue([generateEvent(event, sharedPropsRef, getPage())]);
-      },
-      logEventStart(id, event) {
-        if (!durationEventsQueue.current.has(id)) {
-          durationEventsQueue.current.set(
-            id,
-            generateEvent(event, sharedPropsRef, getPage()),
-          );
-        }
-      },
-      logEventEnd(id, now = new Date()) {
-        const event = durationEventsQueue.current.get(id);
-        if (event) {
-          durationEventsQueue.current.delete(id);
-          event.event_duration =
-            now.getTime() - event.event_timestamp.getTime();
-          if (window.scrollY > 0 && event.event_name !== 'page inactive') {
-            event.page_state = 'active';
-          }
-          pushToQueue([event]);
+  const logEvent = useCallback(
```

Comment:
What's the actual difference between `useCallback` for each function vs `useMemo` for all?

Reply:
With a unique useMemo we force to compute again for each of dependency changes, with independent useCallback we only update the function when the method-specific dependency changes. 

Additionally useCallback is better in this case because it saves specifically the function and not the whole object as result of the hook!

Reply:
Thanks for the explanation!